atlasdb-client/src/main/java/com/palantir/atlasdb/logging/KvsProfilingLogger.java

```diff
@@ -91,36 +92,42 @@ public static void setSlowLogThresholdMillis(long thresholdMillis) {
         slowLogPredicate = createLogPredicateForThresholdMillis(thresholdMillis);
     }
 
-    public static <T> T maybeLog(Supplier<T> action, BiConsumer<LoggingFunction, Stopwatch> logger) {
-        return maybeLog(action, logger, (loggingFunction, result) -> { });
-    }
-
     public static void maybeLog(Runnable runnable, BiConsumer<LoggingFunction, Stopwatch> logger) {
-        maybeLog(() -> {
+        maybeLog((Supplier<Object>) () -> {
             runnable.run();
             return null;
         }, logger);
     }
 
-    public static  <T> T maybeLog(Supplier<T> action, BiConsumer<LoggingFunction, Stopwatch> primaryLogger,
-            BiConsumer<LoggingFunction, T> additonalLoggerWithAccessToResult) {
+    public static <T> T maybeLog(Supplier<T> action, BiConsumer<LoggingFunction, Stopwatch> logger) {
+        return maybeLog(action::get, logger, (loggingFunction, result) -> {});
+    }
+
+    public static <T, E extends Exception> T maybeLog(CallableCheckedException<T, E> action,
+            BiConsumer<LoggingFunction, Stopwatch> primaryLogger) throws E {
+        return maybeLog(action, primaryLogger, ((loggingFunction, result) -> {}));
+    }
+
+    public static <T, E extends Exception> T maybeLog(CallableCheckedException<T, E> action,
+            BiConsumer<LoggingFunction, Stopwatch> primaryLogger,
+            BiConsumer<LoggingFunction, T> additonalLoggerWithAccessToResult) throws E {
         if (log.isTraceEnabled() || slowlogger.isWarnEnabled()) {
             Monitor<T> monitor = Monitor.createMonitor(
                     primaryLogger,
                     additonalLoggerWithAccessToResult,
                     slowLogPredicate);
             try {
-                T res = action.get();
+                T res = action.call();
                 monitor.registerResult(res);
                 return res;
             } catch (Exception ex) {
                 monitor.registerException(ex);
-                throw ex;
+                throw Throwables.throwUncheckedException(ex);
```

Comment:
why are we wrapping this now? we are declaring we throw `E`, but we're actually throwing a `RuntimeException`